CRM/Core/DAO.php

```diff
@@ -957,6 +957,7 @@ public static function deleteRecord(array $record) {
     if (empty($record['id'])) {
       throw new CRM_Core_Exception("Cannot delete {$entityName} with no id.");
     }
+    CRM_Utils_Type::validate($record['id'], 'Positive');
```

Comment:
This is the only thing I moved into the generic function from `Tag::del()` which seems like a reasonable sanity check.